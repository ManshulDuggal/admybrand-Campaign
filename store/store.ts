import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { combineReducers, configureStore } from "@reduxjs/toolkit";
import categoriesSlice from "./reducers/categories";
import countrySlice from "./reducers/country";
import languageSlice from "./reducers/language";
import citySlice from "./reducers/location";
import searchSlice from "./reducers/search";
import userSlice from "./user/userReducer";

const rootReducer = combineReducers({
  categories: categoriesSlice,
  country: countrySlice,
  language: languageSlice,
  location: citySlice,
  search: searchSlice,
  user: userSlice,
});

export const store = configureStore({
  reducer: rootReducer,
});
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
