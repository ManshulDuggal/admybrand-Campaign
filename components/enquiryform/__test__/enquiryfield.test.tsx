import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import EnquiryForm from '../EnquiryForm';


describe("render test", () => {
  it("initial test for enquiry form", () => {
    const component = render(<Provider store={store}>
      <EnquiryForm/>
    </Provider>)
    expect(component).toMatchSnapshot()
  })
})