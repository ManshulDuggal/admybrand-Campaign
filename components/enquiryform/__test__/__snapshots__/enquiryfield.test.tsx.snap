// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test initial test for enquiry form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <button
            style="color: rgb(228, 6, 120); border-bottom-width: 2px; border-bottom-color: #E40678;"
          >
            <h1>
              Enquiry
            </h1>
          </button>
          <button
            style="color: rgb(131, 159, 192); border-bottom-width: 0; border-bottom-color: #839FC0;"
          >
            <h1>
              Refer a Brand
            </h1>
          </button>
        </div>
        <div>
          <form
            action=""
          >
            <div>
              <div>
                <div>
                  <h1>
                    Full Name
                  </h1>
                  <input
                    placeholder="Enter Full Name"
                    type="text"
                  />
                </div>
                <div>
                  <h1>
                    Email
                  </h1>
                  <input
                    placeholder="Enter Email"
                    type="text"
                  />
                </div>
              </div>
              <div>
                <div>
                  <h1>
                    Mobile Number
                  </h1>
                  <input
                    placeholder="Enter Mobile Number"
                    type="text"
                  />
                </div>
                <div>
                  <h1>
                    Alternate Number
                  </h1>
                  <input
                    placeholder="Enter Alternate Number"
                    type="text"
                  />
                </div>
              </div>
              <div>
                <div>
                  <h1>
                    Current City
                  </h1>
                  <input
                    placeholder="Enter Current City"
                    type="text"
                  />
                </div>
                <div>
                  <h1>
                    Franchisee City
                  </h1>
                  <input
                    id=""
                    name=""
                    placeholder="Enter Franchisee City"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div>
              <button>
                <h2>
                  Enquire
                </h2>
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      role="presentation"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUiIGhlaWdodD0iMTUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <noscript />
                  <img
                    alt="enquire"
                    class="object-cover"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <button
          style="color: rgb(228, 6, 120); border-bottom-width: 2px; border-bottom-color: #E40678;"
        >
          <h1>
            Enquiry
          </h1>
        </button>
        <button
          style="color: rgb(131, 159, 192); border-bottom-width: 0; border-bottom-color: #839FC0;"
        >
          <h1>
            Refer a Brand
          </h1>
        </button>
      </div>
      <div>
        <form
          action=""
        >
          <div>
            <div>
              <div>
                <h1>
                  Full Name
                </h1>
                <input
                  placeholder="Enter Full Name"
                  type="text"
                />
              </div>
              <div>
                <h1>
                  Email
                </h1>
                <input
                  placeholder="Enter Email"
                  type="text"
                />
              </div>
            </div>
            <div>
              <div>
                <h1>
                  Mobile Number
                </h1>
                <input
                  placeholder="Enter Mobile Number"
                  type="text"
                />
              </div>
              <div>
                <h1>
                  Alternate Number
                </h1>
                <input
                  placeholder="Enter Alternate Number"
                  type="text"
                />
              </div>
            </div>
            <div>
              <div>
                <h1>
                  Current City
                </h1>
                <input
                  placeholder="Enter Current City"
                  type="text"
                />
              </div>
              <div>
                <h1>
                  Franchisee City
                </h1>
                <input
                  id=""
                  name=""
                  placeholder="Enter Franchisee City"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div>
            <button>
              <h2>
                Enquire
              </h2>
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUiIGhlaWdodD0iMTUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="enquire"
                  class="object-cover"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
